### PR TITLE
EPMRPP-84509 || Project-settings-UI-bugs

### PR DESCRIPTION
**Pull request by BlazarQSO**

### CODE REVIEW

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export const Attribute = ({
   return (
     <div
       ref={wrapperRefCallback}
-      tabIndex={disabled ? -1 : 0}
+      tabIndex={0}
```
Comment (Bam6ycha) [inline app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx:48]: What is the reason for removing `disabled ? -1 : 0`

If the attr is disabled want to be able to focus on this attr.

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss [hunk 2]
```diff
@@ -42,6 +44,10 @@
   max-width: 100%;
   column-gap: 8px;
 
+  &:hover:not(:focus-visible) {
```
Comment (Bam6ycha) [inline app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss:47]: Does this hover state need to appear in the disabled state?

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss [hunk 3]
```diff
@@ -14,7 +14,9 @@
  * limitations under the License.
  */
 
-@mixin draw-attribute-border($width, $height, $translateX: 0, $translateY: 0) {
+@mixin draw-attribute-border($width, $height, $bgColor, $translateX: 0, $translateY: 0) {
+  background-color: $bgColor;
```
Comment (Bam6ycha) [inline app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss:18]: It brakes styles, when you focused on key:value

As I can see the bg color is the same everywhere.

File: app/src/componentLibrary/attributeList/attributeList.scss [hunk 4]
```diff
@@ -23,7 +23,7 @@
 .attributes {
   display: inline-flex;
   flex-wrap: wrap;
-  gap: 16px;
+  gap: 8px;
```
Comment (Bam6ycha) [inline app/src/componentLibrary/attributeList/attributeList.scss:26]: Why do we change the gap between attributes?

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

**Comment by Bam6ycha** [comment]
If you have any additional information from our UX/UI team, do not hesitate and write down the comment in the ticket.

File: app/src/componentLibrary/attributeList/attributeList.scss (hunk 4 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/attributeList/attributeList.scss:26]: according to design

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss (hunk 3 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss]: Done.

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss (hunk 2 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss:45]: if there is focus but the hover should not work

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx (hunk 1 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx:48]: according to design

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx [hunk 5]
```diff
@@ -55,6 +71,8 @@ export const Attribute = ({
         tabIndex={-1}
         className={cx('label', variant, { disabled })}
         onKeyDown={disabled ? null : handleAttributeKeyValueKeyDown}
+        onFocus={onFocus}
```
Comment (Bam6ycha) [inline app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx:74]: It works fine, but only when the key:value in focus.
What about cross icon?

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx [hunk 6]
```diff
@@ -37,16 +38,31 @@ export const Attribute = ({
...
       ref={wrapperRefCallback}
-      tabIndex={disabled ? -1 : 0}
-      className={cx('attribute', variant, customClass, { disabled })}
+      tabIndex={0}
+      className={cx(
+        'attribute',
+        variant,
+        customClass,
+        { disabled },
+        isLabelFocused ? `hover-${variant}` : '',
```
Comment (Bam6ycha) [inline app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx:64]:
```suggestion
{ disabled, [`hover-${variant}`]:isLabelFocused }
```

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.scss]: Done.

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx (hunk 6 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx]: Done.

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx (hunk 5 above)
Comment (BlazarQSO) [reply app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx:71]: Done.

File: app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx [hunk 7]
```diff
@@ -37,16 +38,28 @@ export const Attribute = ({
   handleCrossIconKeyDown,
   wrapperRefCallback,
 }) => {
+  const [isLabelFocused, setIsLabelFocused] = useState(false);
```
Comment (Bam6ycha) [inline app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx:41]: Looks like now, the naming does not match the expected usage.
It is not only about label.

Comment (BlazarQSO) [reply app/src/componentLibrary/attributeList/editableAttribute/attribute/attribute.jsx]: Done.

**Review by Bam6ycha — APPROVED** [review]

**Pull request merged**